src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,99 @@
+    super(INVENTORY_HOLDING_CREATE, vertx, context);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  @Override
+  protected Future<Void> processInventoryCreationEvent(ResourceEvent resourceEvent, String tenantId) {
+    var holdingObject = JsonObject.mapFrom(resourceEvent.getNewValue());
+    var holdingId = holdingObject.getString(InventoryFields.ID.getValue());
+    var dbClient = new DBClient(getVertx(), tenantId);
+    var tenantIdUpdates = List.of(
+      processPoLinesUpdate(holdingId, tenantId, dbClient),
+      processPiecesUpdate(holdingId, tenantId, dbClient)
+    );
+    return GenericCompositeFuture.all(tenantIdUpdates).mapEmpty();
+  }
+
+  private Future<Void> processPoLinesUpdate(String holdingId, String tenantId, DBClient dbClient) {
+    return poLinesService.getPoLinesByHoldingId(holdingId, dbClient)
+      .compose(poLines -> updatePoLines(poLines, holdingId, tenantId, dbClient));
+  }
+
+  private Future<Void> processPiecesUpdate(String holdingId, String tenantId, DBClient dbClient) {
+    return pieceService.getPiecesByHoldingId(holdingId, dbClient)
+      .compose(pieces -> updatePieces(pieces, holdingId, tenantId, dbClient));
+  }
+
+  private Future<Void> updatePoLines(List<PoLine> poLines, String holdingId, String tenantId, DBClient dbClient) {
+    if (CollectionUtils.isEmpty(poLines)) {
+      log.info("updatePoLines:: No poLines to update for holding: '{}' and tenant: '{}'", holdingId, tenantId);
+      return Future.succeededFuture();
+    }
+    log.info("updatePoLines:: Updating {} poLine(s) with holdingId '{}', setting receivingTenantId to '{}'", poLines.size(), holdingId, tenantId);
+    poLines.forEach(poLine -> updateLocationTenantIdIfNeeded(poLine.getLocations(), holdingId, tenantId));
+    return poLinesService.updatePoLines(poLines, tenantId, dbClient)
+      .mapEmpty();
+  }
+
+  private Future<Void> updatePieces(List<Piece> pieces, String holdingId, String tenantId, DBClient client) {
+    var piecesToUpdate = pieces.stream()
+      .filter(piece -> !Objects.equals(piece.getReceivingTenantId(), tenantId))
+      .map(piece -> piece.withReceivingTenantId(tenantId))
+      .toList();
+    if (CollectionUtils.isEmpty(piecesToUpdate)) {
+      log.info("updatePieces:: No pieces to update for holding: '{}' and tenant: '{}'", holdingId, tenantId);
+      return Future.succeededFuture();
+    }
+    log.info("updatePieces:: Updating {} piece(s) with holdingId '{}', setting receivingTenantId to '{}'", pieces.size(), holdingId, tenantId);
+    return pieceService.updatePieces(piecesToUpdate, client);
+  }
+
+  private void updateLocationTenantIdIfNeeded(List<Location> locations, String holdingId, String tenantId) {
+    locations.stream()
+      .filter(location -> Objects.equals(location.getHoldingId(), holdingId))
+      .forEach(location -> location.setTenantId(tenantId));
+  }
+
+  @Override
+  protected Logger getLogger() {
+    return log;
```

Comment:
can we avoid this method?

Reply:
We can create logger in the parent class but it would appear as InventoryCreateAsyncRecordHandler in the logs instead of a concrete child class

Reply:
we can just declare separate loggers in each class